### DISCUSSION TITLE
请问有考虑自定义服务商吗？

### DISCUSSION THREAD

**Discussion by lanbinshijie** [Qexo/Qexo#184] [Ideas]
# 起因

自己在服务器上搭建了Hexo博客（为了方便）
想要接入qexo服务，但发现只能选择GitHub服务商

# 解决方法

请问能否增加选项，添加自己的服务器git服务为服务商？

# 备注

- *服务器与GitHub服务器一样，可以提供repo连接等，token可以换成rsa秘钥*
- *在Issue和Discussion都搜索了相似问题，但未发现相应的疑问。若有重复的，实在抱歉*
- *求回复*

**Comment by am-abudu** [comment]
现在在程序上是支持自行拓展的，如果有能力可以尝试自行添加，或者等待我修复完当前存在的bug后进行

**Reply by lanbinshijie** [reply]
好，谢谢！
PS：好久不见

**Reply by lanbinshijie** [reply]
希望能将自定义服务商提上日程，谢谢qwq
（因为还是很需要的，谢谢啦）

**Comment by am-abudu** [comment]
如果是完全的自定义，只在本地使用的话，问题将变得容易，但如果是Vercel的话，每次都先clone在push极易超时，于是乎你需要什么服务商可以将文档发出来，我进行针对性的适配

**Reply by lanbinshijie** [reply]
嗯嗯，那是否可以考虑新增加一个本地版本呢？比如选择服务商可以换成选择文件夹路径？

**Reply by lanbinshijie** [reply]
因为如果是这样的话，搭建在自己服务器上的博客做到统一适配还是有些困难的，会涉及到git和ssh密钥，而且也不能保证所有人都是这么部署的（可能就我一个），所以考虑本地模式会好一些

**Reply by lanbinshijie** [reply]
不会说话，抱歉了
